quarkus/extensions/src/main/java/org/keycloak/provider/quarkus/QuarkusFilter.java

```diff
@@ -0,0 +1,159 @@
+    public void filter(ContainerRequestContext containerRequestContext) throws IOException {
+
+        if (containerRequestContext.getMediaType() == null) {
+            containerRequestContext.getHeaders().add("Content-Type", MediaType.APPLICATION_FORM_URLENCODED);
+        }
+
+        Resteasy.pushContext(KeycloakApplication.class, keycloakApplication);
+
+        KeycloakSessionFactory sessionFactory = keycloakApplication.getSessionFactory();
+
+        KeycloakSession session = sessionFactory.create();
+
+        Resteasy.pushContext(KeycloakSession.class, session);
+        ClientConnection connection = new ClientConnection() {
+            @Override
+            public String getRemoteAddr() {
+                return request.remoteAddress().host();
+            }
+
+            @Override
+            public String getRemoteHost() {
+                return request.remoteAddress().host();
+            }
+
+            @Override
+            public int getRemotePort() {
+                return request.remoteAddress().port();
+            }
+
+            @Override
+            public String getLocalAddr() {
+                return request.localAddress().host();
+            }
+
+            @Override
+            public int getLocalPort() {
+                return request.localAddress().port();
+            }
+        };
+        session.getContext().setConnection(connection);
+        Resteasy.pushContext(ClientConnection.class, connection);
+
+        KeycloakTransaction tx = session.getTransactionManager();
+
+        Resteasy.pushContext(KeycloakTransaction.class, tx);
+        tx.begin();
+    }
+
+    @Override
+    public void filter(ContainerRequestContext containerRequestContext, ContainerResponseContext containerResponseContext) throws IOException {
+
+        //Work around for https://github.com/vert-x3/vertx-web/issues/1340, or at least something that exhibits the same symptoms
+        //Multiple Set-Cookie headers fail to be returned to the FE, with only 1 making it into the generated response.
+        //However, multiple "NewCookie"'s work but then are issues around quoting the cookie value
+        //See org.keycloak.common.util.ServerCookie on line 183.
+        if (((ContainerResponseContextImpl) containerResponseContext).getHttpResponse().getOutputHeaders().get("Set-Cookie") != null
+                &&  ((ContainerResponseContextImpl) containerResponseContext).getHttpResponse().getOutputHeaders().get("Set-Cookie").size() > 1) {
+            ArrayList<String> list = (ArrayList) ((ContainerResponseContextImpl) containerResponseContext).getHttpResponse().getOutputHeaders().get("Set-Cookie");
+            ArrayList<NewCookie> newCookies = new ArrayList<>();
+            String keycloakSession = "";
+            for (String item: list) {
+                //NewCookie adds quotes around the cookie value for the KEYCLOAK_SESSION (due to /), so skipping this one.
+                if (item.contains("KEYCLOAK_SESSION")) {
```

Comment:
We have this constant already with the cookie name `org.keycloak.services.managers.AuthenticationManager#KEYCLOAK_SESSION_COOKIE`.

I think same goes for `Set-Cookie` which can be grabbed from `javax.ws.rs.core.HttpHeaders#SET_COOKIE`.